libethereum/BlockChain.cpp

```diff
@@ -299,10 +296,17 @@ unsigned BlockChain::open(fs::path const& _path, WithExisting _we)
 #endif
 
 	// TODO: Implement ability to rebuild details map from DB.
-	std::string l;
-	m_extrasDB->Get(m_readOptions, ldb::Slice("best"), &l);
-	m_lastBlockHash = l.empty() ? m_genesisHash : *(h256*)l.data();
-	m_lastBlockNumber = number(m_lastBlockHash);
+	try
+	{
+		auto const l = m_extrasDB->lookup(db::Slice("best"));
+		m_lastBlockHash = *(h256*)l.data();
+		m_lastBlockNumber = number(m_lastBlockHash);
```

Comment:
Can this be done once after the try-catch block instead of duplicating in try and in catch?